tests/testthat/test-model-matrix.R

```diff
@@ -13,3 +13,14 @@ test_that("`model_matrix()` strips all attributes from the `model.matrix()` resu
   # attached "assign" and "contrasts" attributes here
   expect_identical(matrix$Speciessetosa, expect)
 })
+
+test_that("`contr_one_hot()` input checks", {
+  expect_snapshot(contr_one_hot(n = 1, sparse = TRUE))
```

Comment:
`contr_one_hot()` is also in parsnip (where it is exported). 

There are tests there that are a bit more extensive: https://github.com/tidymodels/parsnip/blob/main/tests/testthat/test-contr_one_hot.R

You can bring those over here and I'll open a parsnip issue about the duplication. hardhat would have to export this function for parsnip to use it.

Reply:
tidymodels/parsnip#1215